MediaGalleryRenditions/Model/GetRenditionPath.php

```diff
@@ -0,0 +1,135 @@
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Exception\FileSystemException;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Filesystem;
+use Magento\Framework\Filesystem\DriverInterface;
+use Magento\Framework\Filesystem\Io\File;
+use Magento\MediaGalleryRenditionsApi\Api\GetRenditionPathInterface;
+
+class GetRenditionPath implements GetRenditionPathInterface
+{
+    /**
+     * @var Filesystem\Directory\WriteInterface
+     */
+    private $directory;
+
+    /**
+     * @var File|mixed|null
+     */
+    private $ioFile;
+
+    /**
+     * @var DriverInterface|mixed|null
+     */
+    private $file;
+
+    private const RENDITIONS_DIRECTORY_NAME = '.renditions';
+
+    /**
+     * @param Filesystem $filesystem
+     * @param File|null $ioFile
+     * @param DriverInterface|null $file
+     * @throws FileSystemException
+     */
+    public function __construct(
+        Filesystem $filesystem,
+        File $ioFile = null,
+        DriverInterface $file = null
+    ) {
+        $this->directory = $filesystem->getDirectoryWrite(DirectoryList::MEDIA);
+        $this->ioFile = $ioFile ?: ObjectManager::getInstance()->get(File::class);
+        $this->file = $file ?: ObjectManager::getInstance()->get(Filesystem\Driver\File::class);
+    }
+
+    /**
+     * Returns Rendition image path
+     *
+     * @param string $path
+     * @return string
+     * @throws LocalizedException
+     * @throws FileSystemException
+     */
+    public function execute(string $path) :string
+    {
+        $realPath = $this->directory->getRelativePath($path);
+        if (!$this->directory->isFile($realPath) || !$this->directory->isExist($realPath)) {
+            throw new LocalizedException(__('Directory or File %1 does not exist in media directory.', $realPath));
+        }
+        $renditionImageDirectoryPath = $this->getRenditionsImageDirectory($path);
+        $renditionImageDirectory = $this->directory->getRelativePath($renditionImageDirectoryPath);
+        if (!$this->directory->isExist($renditionImageDirectory)) {
+            $this->directory->create($renditionImageDirectory);
+        }
+        if (!$this->directory->isExist($renditionImageDirectory)) {
+            throw new LocalizedException(__(
+                'Directory %1 does not exist in media directory.',
+                $renditionImageDirectory
+            ));
+        }
+        return $renditionImageDirectoryPath . '/' . $this->ioFile->getPathInfo($path)['basename'];
+    }
+
+    /**
+     * Return renditions directory path for file/current directory
+     *
+     * @param bool|string $filePath Path to the file
+     * @return string
+     */
+    private function getRenditionsImageDirectory($filePath = false) :string
+    {
+        $renditionRootDir = $this->getRenditionsRoot();
+
+        if ($filePath) {
+            $renditionImagePath = $this->getRenditionsPath($filePath, false);
+            if ($renditionImagePath) {
```

Comment:
By having getRenditionsPath throwing an exception we will be able to avoid this if.